sw/device/lib/crypto/impl/ecc.c

```diff
@@ -12,25 +14,48 @@
 crypto_status_t otcrypto_ecdsa_keygen(const ecc_curve_t *elliptic_curve,
                                       crypto_blinded_key_t *private_key,
                                       ecc_public_key_t *public_key) {
-  // TODO: Connect ECDSA operations to API.
-  return kCryptoStatusNotImplemented;
+  crypto_status_t err =
+      otcrypto_ecdsa_keygen_async_start(elliptic_curve, private_key->config);
+  // TODO(#17803): replace this error check with HARDENED_TRY if cryptolib and
+  // status_t errors become more compatible.
+  if (launder32(err) != kCryptoStatusOK) {
+    return err;
+  }
+  HARDENED_CHECK_EQ(err, kCryptoStatusOK);
+  return otcrypto_ecdsa_keygen_async_finalize(elliptic_curve, private_key,
+                                              public_key);
 }
 
 crypto_status_t otcrypto_ecdsa_sign(const crypto_blinded_key_t *private_key,
```

Comment:
As part of the requirements, will we have to support any utilities to convert `signtature` into ASN1 format, or is that outside the scope of cryptolib? (asking because of interop with host tooling. I am ok if we don't support it), although we may have to for attestation - we could use a binary template if needed?

Reply:
Encoding/decoding ASN1 is outside the scope of cryptolib in the current plans. For attestation it's probably specific enough that we can do something special-case, like you suggest.